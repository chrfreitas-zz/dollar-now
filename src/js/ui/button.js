import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types'

const ButtonStyled = styled.button`
    background: transparent;
    border: 1px solid #fff;
    border-radius: 3px;
    color: #fff;
    padding: 5px 30px;
    text-transform: uppercase;
    cursor: pointer;

    &:disabled {
        opacity: .2;
    }
`;

class Button extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <ButtonStyled>
                { this.props.children }
            </ButtonStyled>
        )
    }
}

Button.propTypes = {
    children: PropTypes.any
}

export { Button, ButtonStyled}
