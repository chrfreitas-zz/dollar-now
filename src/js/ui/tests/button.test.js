import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import { Button } from './../button';

beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
});

describe('Button', () => {
    it('should return true', () => {
        const ButtonWrapper = shallow(<Button>Hi!</Button>);
        expect(ButtonWrapper.exists()).toBeTruthy();
    });

    it('should return Hi! when pass like children', () => {
        const ButtonWrapper = shallow(<Button>Hi!</Button>);
        expect(ButtonWrapper.prop('children')).toBe("Hi!");
    });

    it('should match with snapshot', () => {
        const ButtonSnapshot = renderer.create(<Button>Hi!</Button>);
        expect(ButtonSnapshot).toMatchSnapshot();
    })
})


