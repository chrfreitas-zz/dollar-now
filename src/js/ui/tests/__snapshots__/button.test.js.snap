// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match with snapshot 1`] = `
<button
  className="sc-bdVaJa lnmrTy"
>
  Hi!
</button>
`;
